docs/data/sdks/javascript/index.md

```diff
@@ -983,23 +983,86 @@ var exitCallback = function {
 amplitude.getInstance().init('API_KEY', 'USER_ID', { onExitPage: exitCallback });
 ```
 
-### Custom device IDs
+### Device ID lifecycle
 
-By default, device IDs are randomly generated UUIDs. You can define a custom device ID by setting it as a configuration option or by calling `setDeviceId`.
+By default Amplitude Javascript SDK uses a base64id as device id in order to have more compact string IDs. A 22-character base64id has the same range as a 128-bit UUID, which is represented as a 36 character string.
+
+On instance initialization, the SDK initializes the device ID by the following order:
```

Comment:
I feel we need to emphasize when it gets a value, it returns and ignores the others. Also at the end, we should add 4. base64id()

How about this: 
```During instance initialization, the SDK initializes the device ID based on a predefined order. The deviceId will be chosen when it encounters a value, prioritizing the first valid value it finds in the order.```

Reply:
Thanks for your review Alyssa. It is more clear to emphasize that the device id will be the first valid value. 

Fix.